sklearn/inspection/_partial_dependence.py

```diff
@@ -954,3 +1111,470 @@ def plot(self, ax=None, n_cols=3, line_kw=None, contour_kw=None):
                 axi.set_ylabel(self.feature_names[fx[1]])
                 axi.set_xlim(xlim)
         return self
+
+
+def _ice_brute(estimator, grid, features_indices, X, response_method):
+    predictions = _get_predictions(estimator, grid, features_indices, X,
+                                   response_method)
+    instances = X.shape[0]
+
+    # reshape to (n_targets, n_instances, n_points) where n_targets is:
+    # - 1 for non-multioutput regression and binary classification (shape is
+    #   already correct in those cases)
+    # - n_tasks for multi-output regression
+    # - n_classes for multiclass classification.
+    predictions = np.array(predictions).T
+    if is_regressor(estimator) and predictions.ndim == 2:
+        # non-multioutput regression, shape is (n_instances, n_points,)
+        predictions = predictions.reshape(instances, -1)
+    elif is_classifier(estimator) and predictions.shape[0] == 2:
+        # Binary classification, shape is (2, n_instances, n_points).
+        # we output the effect of **positive** class
+        predictions = predictions[1]
+        predictions = predictions.reshape(instances, -1)
+
+    return predictions
+
+
+def individual_conditional_expectation(estimator, X, features,
```

Comment:
I think that we could add a useful parameter `subsample`. Basically, it will not be easy to visualize all responses and we need to subsample anyway. So we can add the parameter directly.